packages/celotool/src/lib/testnet-utils.ts

```diff
@@ -22,29 +23,52 @@ export async function uploadStaticNodesToGoogleStorage(networkName: string) {
   console.info(`\nUploading static nodes for ${networkName} to Google cloud storage...`)
   // Get node json file
   let nodesJsonData: string | null = null
-  const numAttempts = 100
-  for (let i = 1; i <= numAttempts; i++) {
+  nodesJsonData = await tryNTimes(
+    async () => JSON.stringify(await getEnodesWithExternalIPAddresses(networkName)),
+    100
+  )
+  if (nodesJsonData === null) {
+    throw new Error('Fail to get static nodes information')
+  }
+  console.debug('Static nodes are ' + nodesJsonData + '\n')
+  const localTmpFilePath = `/tmp/${networkName}_static-nodes`
+  writeFileSync(localTmpFilePath, nodesJsonData)
+  await uploadFileToGoogleStorage(localTmpFilePath, staticNodesBucketName, networkName, true)
+}
+
+// only intended to be used for VM testnets because non-VM testnets do not support
+// discovery v5 due to udp IP issues
+export async function uploadBootnodesToGoogleStorage(celoEnv: string) {
+  console.info(`\nUploading bootnodes for ${celoEnv} to Google cloud storage...`)
+  const bootnodeEnodesJson = await tryNTimes(
+    async () => JSON.stringify(await getBootnodeEnodes(celoEnv)),
+    100
+  )
+  if (!bootnodeEnodesJson) {
+    throw new Error('Failed to get bootnode enodes')
+  }
+  console.debug('Bootnode enodes are ' + bootnodeEnodesJson + '\n')
+  const localTmpFilePath = `/tmp/${celoEnv}_bootnodes`
+  writeFileSync(localTmpFilePath, bootnodeEnodesJson)
+  await uploadFileToGoogleStorage(localTmpFilePath, bootnodesBucketName, celoEnv, true)
+}
+
+async function tryNTimes(action: () => Promise<any>, attempts: number) {
+  for (let i = 1; i <= attempts; i++) {
     try {
-      nodesJsonData = JSON.stringify(await getEnodesWithExternalIPAddresses(networkName))
-      break
+      const result = await action()
+      return result
     } catch (error) {
       const sleepTimeBasisInMs = 1000
       const sleepTimeInMs = sleepTimeBasisInMs * Math.pow(2, i)
```

Comment:
nit: That is going to be a pretty huge timeout! consider capping the timeout. Or maybe not call it with `attempts = 100` ?